lib/foreman/renderer/base_renderer.rb

```diff
@@ -7,6 +7,7 @@ def initialize(source, scope)
         @source = source
         @scope = scope
         @scope.instance_variable_set('@template_name', source_name)
+        @scope.instance_variable_set('@source', @source)
```

Comment:
if we're linking the two, should we rather do it where we create the scope? the scope initializer could accept source as another keyword, in `Template#render` we could simply first create source and pass it to scope. This could also replace the line above, the scope initializer would load template name out of source. This way we'd get rid of `instance_variable_set`